packages/bloc_test/test/bloc_test_test.dart

```diff
@@ -86,6 +98,18 @@ void main() {
         },
         expect: [1, 2],
       );
+
+      blocTest(
+        'emits [1, 2] when AsyncCounterEvent.increment is called multiple'
+        'times with async act and expectAsync',
+        build: () async => AsyncCounterBloc(),
+        act: (bloc) async {
+          bloc.add(AsyncCounterEvent.increment);
+          await Future.delayed(Duration(milliseconds: 10));
+          bloc.add(AsyncCounterEvent.increment);
+        },
+        expectAsync: () async => [Future.value(1), 2],
```

Comment:
I don't really understand this. The bloc should not be emitting Futures can you please explain why this is needed?

Reply:
to test if the future returns before comparing the expected array with the actual states.

Reply:
but the way `blocTest` works is it will wait until all the states are emitted before calling `expect`